atom/common/node_bindings.cc

```diff
@@ -119,9 +124,25 @@ NodeBindings::~NodeBindings() {
   uv_sem_destroy(&embed_sem_);
   uv_close(reinterpret_cast<uv_handle_t*>(&dummy_uv_handle_), nullptr);
 
-  // Destroy loop.
-  if (uv_loop_ != uv_default_loop())
-    uv_loop_delete(uv_loop_);
+  // Clean up worker loop
+  if (uv_loop_ == &worker_loop_) {
+    // Close any active handles
+    uv_stop(uv_loop_);
+    uv_walk(uv_loop_, [](uv_handle_t* handle, void*){
+      if (!uv_is_closing(handle)) {
+        uv_close(handle, nullptr);
+      }
+    }, nullptr);
+
+    // Run the loop to let it finish all the closing handles
+    // NB: after uv_stop(), uv_run(UV_RUN_DEFAULT) returns 0 when that's done
+    for (;;)
+      if (!uv_run(uv_loop_, UV_RUN_DEFAULT))
+        break;
+
+    DCHECK_EQ(uv_loop_alive(uv_loop_), 0);
```

Comment:
`DCHECK(!uv_loop_alive(...))` maybe?

Reply:
Sure. Committed in 26ace8ddae1a29c08890115553e98bbc67cd405e